traefik/faency
# chore: semantic release dry run

## Description

PR split from:
- https://github.com/traefik/faency/pull/288

Adds a GH action to trigger a "dry run" of semantic release during PR review

This is an experimental proposal to run the "dry run mode" of semantic release as a github action and output the release notes inside a PR, to allow reviewers to discuss the release.

1. There's a run which checks against all commits of a PR, to show devs what version would be expected from their commit messages.
2. There's a run which checks only against a simulation of the "squash and merge" behaviour of github PRs

The goal of this experiment is to show all capabilities of the toolkit we are studying, and spark a discussion around a possible usage.

## Review expected

This PR is peculiar in the sense that I don't really expect to merge it as-is.

The workflow I implemented is not battled-tested, nor optimized. It's more of a POC. 

I'm even wondering if my idea brings value.

I expect your review to revolve around:
- improvements
- alternatives
- similar ideas
- usefulness




Preset: [conventional-commits](https://www.conventionalcommits.org/en/v1.0.0/)

## Dependency changes



| Dependency | Version        | State   |
| ---------- | -------------- | ------- |
| [semantic-release](https://github.com/semantic-release/semantic-release)       | ^19.0.2          | Added   |
| [@semantic-release/commit-analyzer](https://github.com/semantic-release/commit-analyzer)       | ^9.0.2          | Added   |
| [@semantic-release/github](https://github.com/semantic-release/github)       | ^8.0.2          | Added   |
| [@semantic-release/npm](https://github.com/semantic-release/npm)       | ^9.0.0          | Added   |
| [@semantic-release/release-notes-generator](https://github.com/semantic-release/release-notes-generator)       | ^10.0.3          | Added   |
| [conventional-changelog-conventionalcommits](https://github.com/conventional-changelog/conventional-changelog)       | ^4.6.3          | Added   |

- [semantic-release](https://semantic-release.gitbook.io/semantic-release) is responsible for auto-releasing PRs, following plugins are the default ones required to handle the full feature:
  - @semantic-release/commit-analyzer
  - @semantic-release/github
  - @semantic-release/npm
  - @semantic-release/release-notes-generator
- [conventional-changelog-conventionalcommits](https://www.conventionalcommits.org/en/v1.0.0/) is the preset we use for the commit analyzer to compute the proper release version, based on commit message merged on master


## How to test

The GH action will be able to run only once we merge this PR.

You can play with our test repo with the dedicated PR:
- https://github.com/matthieuh/test-semantic-release/pull/7

Please note that because of our fork-PR strategy, we cannot use `pull_request` GH event to trigger the workflow, whereas it is possible (for the sake of playground) on the test repo
